cuda/solver/batch_cg_kernels.cu

```diff
@@ -74,70 +74,54 @@ namespace batch_cg {
 template <typename T>
 using BatchCgOptions = gko::kernels::batch_cg::BatchCgOptions<T>;
 
+#define BATCH_CG_KERNEL_LAUNCH(_stoppertype, _prectype)                      \
+    apply_kernel<stop::_stoppertype<ValueType>>                              \
+        <<<nbatch, default_block_size, shared_size>>>(                       \
+            opts.max_its, opts.residual_tol, logger, _prectype<ValueType>(), \
+            a, b, x)
+
 template <typename BatchMatrixType, typename LogType, typename ValueType>
-static void apply_impl(
-    std::shared_ptr<const CudaExecutor> exec,
-    const BatchCgOptions<remove_complex<ValueType>> opts, LogType logger,
-    const BatchMatrixType &a,
-    const gko::batch_dense::UniformBatch<const ValueType> &left,
-    const gko::batch_dense::UniformBatch<const ValueType> &right,
-    const gko::batch_dense::UniformBatch<ValueType> &b,
-    const gko::batch_dense::UniformBatch<ValueType> &x)
+static void apply_impl(std::shared_ptr<const CudaExecutor> exec,
+                       const BatchCgOptions<remove_complex<ValueType>> opts,
+                       LogType logger, const BatchMatrixType &a,
+                       const gko::batch_dense::UniformBatch<const ValueType> &b,
+                       const gko::batch_dense::UniformBatch<ValueType> &x)
 {
     using real_type = gko::remove_complex<ValueType>;
     const size_type nbatch = a.num_batch;
 
+    int shared_size =
+#if GKO_CUDA_BATCH_USE_DYNAMIC_SHARED_MEM
+        gko::kernels::batch_cg::local_memory_requirement<ValueType>(a.num_rows,
+                                                                    b.num_rhs);
+#else
+        0;
+#endif
 
     if (opts.preconditioner == gko::preconditioner::batch::type::none) {
-        const int shared_size =
+        shared_size +=
```

Comment:
I think you are missing the `#else` here

Reply:
Good catch, the `shared_size +=` should be inside the conditional.